File: RafJira/app/src/main/java/com/example/view/fragments/MainFragment.java
package com.example.view.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

import com.example.raf_jira.R;
import com.example.raf_jira.databinding.FragmentMainBinding;
import com.example.view.viewPager.PagerAdapterMainFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainFragment extends Fragment {

    private FragmentMainBinding binding;
    private ViewPager viewPager;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentMainBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init();
    }

    private void init() {
        initViewPager();
        initNavigation();
    }

    private void initViewPager() {
        viewPager = binding.viewPager;
        viewPager.setAdapter(new PagerAdapterMainFragment(getParentFragmentManager()));
    }

    private void initNavigation() {
        ((BottomNavigationView)binding.bottomNavigation).setOnItemSelectedListener(item -> {
            switch (item.getItemId()) {
                // setCurrentItem metoda viewPager samo obavesti koji je Item trenutno aktivan i onda metoda getItem u adapteru setuje odredjeni fragment za tu poziciju
                case R.id.navigation_1: viewPager.setCurrentItem(PagerAdapterMainFragment.FRAGMENT_1, false); break;
                case R.id.navigation_2: viewPager.setCurrentItem(PagerAdapterMainFragment.FRAGMENT_2, false); break;
                case R.id.navigation_3: viewPager.setCurrentItem(PagerAdapterMainFragment.FRAGMENT_3, false); break;
                case R.id.navigation_4: viewPager.setCurrentItem(PagerAdapterMainFragment.FRAGMENT_4, false); break;
            }
            return true;
        });
    }
}

File: RafTrading/app/src/main/java/com/example/raftrading/features/navigation/MainFragment.kt
package com.example.raftrading.features.navigation

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import com.example.raftrading.R
import com.example.raftrading.databinding.FragmentMainBinding
import com.example.raftrading.features.navigation.viewPager.PagerAdapterMainFragment

class MainFragment: Fragment() {

    private lateinit var binding: FragmentMainBinding
    private lateinit var viewPager: ViewPager

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentMainBinding.inflate(layoutInflater)
        return  binding.root
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    private fun init() {
        initViewPager()
        initNavigation()
    }

    private fun initViewPager() {
        viewPager = binding.viewPager
        viewPager.adapter = PagerAdapterMainFragment(childFragmentManager)
    }

    private fun initNavigation() {
        binding.bottomNavigation.setOnItemSelectedListener { item ->
            when (item.itemId) {
                R.id.navigation_1 -> viewPager.setCurrentItem(PagerAdapterMainFragment.FRAGMENT_1,false)
                R.id.navigation_2 -> viewPager.setCurrentItem(PagerAdapterMainFragment.FRAGMENT_2,false)
                R.id.navigation_3 -> viewPager.setCurrentItem(PagerAdapterMainFragment.FRAGMENT_3,false)
            }
            true
        }
    }

}